packages/filler/src/services/ContractInteractionService.ts

```diff
@@ -399,4 +399,59 @@ export class ContractInteractionService {
 		})
 		return hostParams
 	}
+
+	async getTokenUsdValue(order: Order): Promise<{ outputUsdValue: bigint; inputUsdValue: bigint }> {
+		let outputUsdValue = BigInt(0)
+		let inputUsdValue = BigInt(0)
+		const outputs = order.outputs
+		const inputs = order.inputs
+
+		for (const output of outputs) {
+			const tokenAddress = bytes32ToBytes20(output.token)
+			const amount = output.amount
+			const price = await this.getTokenPrice(tokenAddress, order.destChain)
+			outputUsdValue = outputUsdValue + amount * price
+		}
+
+		for (const input of inputs) {
+			const tokenAddress = bytes32ToBytes20(input.token)
+			const amount = input.amount
+			const price = await this.getTokenPrice(tokenAddress, order.sourceChain)
+			inputUsdValue = inputUsdValue + amount * price
+		}
+
+		return { outputUsdValue, inputUsdValue }
+	}
+
+	async getTokenPrice(tokenAddress: string, chain: string): Promise<bigint> {
+		const decimals = await this.getTokenDecimals(tokenAddress, chain)
+		const usdValue = await fetchTokenUsdPriceOnchain(tokenAddress, decimals)
+		return usdValue
+	}
+
+	async getFillerBalanceUSD(order: Order, chain: string): Promise<bigint> {
+		// As part of the protocol, the filler will have only two tokens:
+		// 1. The native token of the chain. And 2. DAI
```

Comment:
I am expecting the filler to hold only two tokens. Native and Dai. Is my assumption correct? @seunlanlege @Wizdave97

Reply:
They can probably hold USDC or USDT also; those are stablecoins with high availability.